### PR TITLE
feat: add github actions to build 

### LINKED ISSUE #4: add github actions to build and lint
@michaeleggers i probably need your help for the windows build. as far as i saw it we can choose the platform in the actions without problems :) 

- [ ] setup linting on pull request
- [ ] setup linux builds and save the artifacts
- [ ] setup windows builds and save the artifacts

### PR DESCRIPTION
**Pull request by benekoehler**
Closes #4

### CODE REVIEW

**Review by michaeleggers — COMMENTED** [review]
Thanks!! This is really useful. 
- Can we somehow specify a concrete version of the compilers (clang/++)? I am not sure if this even makes sense but I have witnessed enough weird bugs due to c++ compiler differences in the past. It might be useful that we all build with the exact same version.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  #        run: ./tools/lint.sh
+  #      - name: Run clang-format
+  #        run: ./tools/format.sh
+
+  build:
+    runs-on: ubuntu-latest
+    #    needs: lint
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (Cameramorphic) [inline .github/workflows/build.yaml:53]: There are warnings about deprecated node versions with this action. Is there any reason against using the latest `actions/checkout@v4` ?

Comment (benekoehler) [reply .github/workflows/build.yaml:53]: absolutly not :)

**Review by michaeleggers — APPROVED** [review]
Thanks for forcing clang 18. As discusses with @benekoehler we haven't found a way to specify the exact version so if the Linux people work on clang 18.x that should do it for now.

**Pull request merged**